src/swap/hw-app-swap/Swap.js

```diff
@@ -0,0 +1,211 @@
+// @flow
+import type Transport from "@ledgerhq/hw-transport";
+
+const START_NEW_TRANSACTION_COMMAND: number = 0x01;
+const SET_PARTNER_KEY_COMMAND: number = 0x02;
+const CHECK_PARTNER_COMMAND: number = 0x03;
+const PROCESS_TRANSACTION_RESPONSE: number = 0x04;
+const CHECK_TRANSACTION_SIGNATURE: number = 0x05;
+const CHECK_PAYOUT_ADDRESS: number = 0x06;
+const CHECK_REFUND_ADDRESS: number = 0x07;
+
+export default class Swap {
+  transport: Transport<*>;
+  allowedStatuses: Array<number> = [
+    0x9000,
+    0x6a80,
+    0x6a81,
+    0x6a82,
+    0x6a83,
+    0x6a84,
+    0x6a85,
+    0x6e00,
+    0x6d00,
+    0x9d1a
+  ];
+
+  constructor(transport: Transport<*>) {
+    this.transport = transport;
+  }
+
+  isSuccess(result: Buffer): boolean {
+    return (
+      result.length >= 2 && result.readUInt16BE(result.length - 2) === 0x9000
+    );
+  }
+
+  mapProtocolError(result: Buffer): void {
+    if (result.length < 2) throw new Error("Response length is too small");
+    var errorMessage: string;
+    switch (result.readUInt16BE(result.length - 2)) {
```

Comment:
https://github.com/LedgerHQ/ledgerjs/blob/master/packages/errors/src/index.ts#L300

here, we simply should `throw new TransportStatusError(result.readUInt16BE(result.length - 2))` to replace this switch code.